### ISSUE TITLE
Describe the optimization performed the models during training

### ISSUE THREAD

**Issue by rafaloz** [autogluon/autogluon#1848]
### Describe the issue linked to the documentation

I am working towards a regression model for age prediction from MRI features. I am using all continuous features. In the literature, SVR is usually the regressor used for its performance and ability to generalize. Nevertheless, I tested your library on my data and checked that the NN_TORCH model performs way better than the SVR on my data. I have been reviewing the process for training the model, however, the documentation seems a bit confusing and lacks detail for what I want to understand. My aim is to get all the details about the training of the network, which hyperparameters are you testing, why are you selecting the ones selected, etc. Also, I have checked the code, but going step by step over your code is a gargantuan effort, mainly because the code is extremely professional and performs checks for all the situations possible, which in fact is great.

### Suggest a potential alternative/fix

Would it be possible to have a step-by-step process of the optimization procedure for the NN_TORCH model? data preprocessing, initialization, the hyperparameters tested...

I have been able to check that: 

1.- You standardize continuous features and normalize with a power transform features with a skewed distribution. 
2.- You use an MLP with 128 nodes and 4 layers as the first NN_torch model to train
5.- You add weight decay as regularization
3.- you separate data in train and validation and stop training if after 20 epochs there is no improvement in validation (training up to 500 epochs)
4.- You divide the training data into 6 batches (at least with my data) 

I built my own MLP and checked that your loss at the beginning of the training is way better than mine. Are you performing any initialization procedure beyond PyTorch's "reset_parameters"? Could it be explained just by a lucky random initialization? 

Thank you for your work! This library is a great tool!

**Comment by Innixma** [comment]
Hi @rafaloz,

Unfortunately, providing a detailed explanation is challenging to say the least.

As to what we do and why we did it:

https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-faq.html#how-can-i-reproduce-the-results-of-model-xyz-without-using-autogluon

https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-faq.html#why-was-xyz-design-decision-chosen

**Comment by rafaloz** [comment]
Hello @Innixma, 

After some tinkering, I checked that the early stopping (point 3) seems to be the element that most improves performance. Regularization is key! 

Thanks for the follow-up. 

Rafa
